src/UhdmAst.cpp

```diff
@@ -153,8 +154,14 @@ AstNode* get_value_as_node(vpiHandle obj_h) {
     }
     case vpiRealVal: {
         std::string valStr = std::to_string(val.value.real);
-        V3Number value(value_node, valStr.c_str());
-        value_node = new AstConst(new FileLine("uhdm"), value);
+	
+        bool parseSuccess;
+        double value = VString::parseDouble(valStr, &parseSuccess);
+        assert(parseSuccess && "Unable to parse real value");
```

Comment:
Please use a Verilator-style assert:
https://github.com/antmicro/verilator-1/blob/53b9c4fc44c397c39d2c90b0b3b12e4efc43a865/src/V3Width.cpp#L372

Reply:
Thank you, done